src/core/public/overlays/flyout/flyout_service.tsx

```diff
@@ -84,6 +84,8 @@ export interface OverlayFlyoutOpenOptions {
   'data-test-subj'?: string;
   size?: EuiFlyoutSize;
   maxWidth?: boolean | number | string;
+  hideCloseButton?: boolean;
+  onClose?: (flyout: OverlayRef) => void;
```

Comment:
Would you mind adding a comment that `flyout.close` must be called by an external component?

Reply:
Sure 👍